lib/os/Kconfig

```diff
@@ -125,6 +125,13 @@ config UTF8
 	  Enable the utf8 API. The API implements functions to specifically
 	  handle UTF-8 encoded strings.
 
+config ASYNC
```

Comment:
do not like this, it will end up as CONFIG_ASYNC and that can mean a lot of things, the symbol should be more targeted and should provide context in the name.